firefox_puppeteer/ui/windows.py

```diff
@@ -349,7 +349,8 @@ def window_opened(mn):
 
         window = self._windows.create_window_instance(new_handle, expected_window_class)
 
-        Wait(self.marionette).until(lambda _: window.focused == expect_focus)
+        if expect_focus:
+            window.focus()
```

Comment:
I just noticed the discrepancy of the parameter name and our action. Right now we indeed only check for an expected state. So that is correct. But it looks like (as you found out) that the new window is not getting correctly focused. It might be a Firefox or a Marionette issue. 

What if we rename the parameter to `focus` and still let it default to `True`. All other modules would have to be checked to ensure we don't break anything. But then it's clear what we want here.